lib/rubocop/cop/rspec/expectation_target_method.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Checks ExpectationTarget method.
+      #
+      # @example
+      #   # bad
+      #   expect(something).kind_of? Foo
+      #   is_expected == 42
+      #
+      #   # good
+      #   expect(something).to be_a Foo
+      #   is_expected.to eq 42
+      #
+      class ExpectationTargetMethod < Base
+        MSG = 'Do not use any method other than `.to` or `.not_to`.'
+        RESTRICT_ON_SEND = %i[expect is_expected].freeze
+
+        # @!method expect?(node)
+        def_node_matcher :expect?, <<~PATTERN
+          {
+            (send nil? :expect ...)
+            (send nil? :is_expected)
+          }
+        PATTERN
+
+        # @!method expect_block?(node)
+        def_node_matcher :expect_block?, <<~PATTERN
+          (block #expect? (args) _body)
+        PATTERN
+
+        # @!method expectation_target_method?(node)
+        def_node_matcher :expectation_target_method?, <<~PATTERN
+          (send _ #Runners.all ...)
+        PATTERN
+
+        def on_send(node)
+          return unless expect?(node)
+          return if node.parent.block_type?
+
+          check_expect(node)
+        end
+
+        def on_block(node) # rubocop:disable InternalAffairs/NumblockHandler
+          return unless expect_block?(node)
+
+          check_expect(node)
```

Comment:
Can you please add some coverage for this? Do we even need on_block?

Reply:
on_block is defined to support the `expect` with block. The test is below.
https://github.com/rubocop/rubocop-rspec/pull/1893/files#diff-5e17efe314d6a76ce15c21be790197c8d4958d545fbb923dc78881b4172f2a3fR36

This was implemented with reference to [void_expect.rb](https://github.com/rubocop/rubocop-rspec/blob/13538ffa10bc819dfaaba89d81e702dc82a90ea1/lib/rubocop/cop/rspec/void_expect.rb#L36).

If I check the node type of the parent, it looks like only `on_send` would be fine, is that better?
The following is an image of the implementation using only on_send.
```rb
        def on_send(node)
          return unless expect?(node)

          if node.parent&.block_type?
            check_expect(node.parent)
          else
            check_expect(node)
          end
        end
```

Reply:
Thanks for the explanation, it makes total sense to check the block syntax, too.
But it seems that with no regards to the presence or absence of the block it can be detected in on_send, can’t it?

Reply:
I removed on_block because I could detect it only with on_send.